### PR TITLE
DSND-1451: Refactor company appointments api to call metrics api

### PR DESCRIPTION
**Pull request by crowlandsCH**
* Makes a call to company-metrics-api to populate counts returned in response.
* Sets inactive count based on company status from first appointment in result set.

[DSND-1451](https://companieshouse.atlassian.net/browse/DSND-1451)

[DSND-1451]: https://companieshouse.atlassian.net/browse/DSND-1451?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/company_appointments/service/CompanyRegisterService.java [hunk 1]
```diff
@@ -1,38 +1,30 @@
...
-    @Autowired
-    public CompanyRegisterService(CompanyMetricsApiService companyMetricsApiService) {
-        this.companyMetricsApiService = companyMetricsApiService;
-    }
+    private static final String PUBLIC_REGISTER = "public-register";
 
-    public boolean isRegisterHeldInCompaniesHouse (String registerType, String companyNumber) throws ServiceUnavailableException, BadRequestException {
-        RegistersApi registersApi = companyMetricsApiService.invokeGetMetricsApi(companyNumber).getData().getRegisters();
+    public boolean isRegisterHeldInCompaniesHouse(String registerType, RegistersApi registersApi)
+            throws BadRequestException {
```
Comment (johncook-ch) [inline src/main/java/uk/gov/companieshouse/company_appointments/service/CompanyRegisterService.java:13]: `BadRequestException` is a `RuntimeException` and so does not need to be declared in the `throws` clause

**Review by johncook-ch — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/companieshouse/company_appointments/service/CompanyRegisterService.java (hunk 1 above)
Comment (crowlandsCH) [reply src/main/java/uk/gov/companieshouse/company_appointments/service/CompanyRegisterService.java]: Fixed in latest commit

**Review by johncook-ch — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/company_appointments/service/CompanyAppointmentService.java [hunk 2]
```diff
@@ -108,18 +116,14 @@ public OfficerList fetchAppointmentsForCompany(FetchAppointmentsRequest request)
...
-        } else {
-            activeCount = count;
-        }
+        String companyStatus = allAppointmentData.get(0).getCompanyStatus();
 
-        int resignedCount = (int) officerSummaries.stream().filter(officer -> officer.getResignedOn() != null && officer.getResignedOn().isBefore(ChronoLocalDate.from(LocalDate.now().atStartOfDay()))).count();
+        Counts counts = Optional.ofNullable(metricsApi.getCounts())
+                .map(CountsApi::getAppointments)
+                .map(appointments -> new Counts(appointments, companyStatus))
+                .orElseThrow(() -> new NotFoundException(String.format("Appointments for company [%s] not found", companyNumber)));
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/company_appointments/service/CompanyAppointmentService.java:124]: Perhaps we should change this message to something like "Appointments metrics for company number not found"?

**Review by cgregoryCH — COMMENTED** [review]
Line 83 of CompanyAppointmentService.java can also have the BadRequestException declaration removed.

File: src/main/java/uk/gov/companieshouse/company_appointments/service/CompanyAppointmentService.java (hunk 2 above)
Comment (crowlandsCH) [reply src/main/java/uk/gov/companieshouse/company_appointments/service/CompanyAppointmentService.java:124]: Done

**Review by kinpang-CH — APPROVED** [review]

**Review by sthompsonCH — APPROVED** [review]

**Review by cgregoryCH — APPROVED** [review]

**Pull request merged**